### PR TITLE
Refactors sessions

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Refactors sessions into new queries folder

## Checklist

- [x] All querysets/queries filter by Team (if applicable)
- [x] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

File: posthog/api/event.py [hunk 1]
```diff
@@ -261,274 +256,23 @@ def values(self, request: request.Request) -> response.Response:
...
+        filter = Filter(request=request)
+        result: Dict[str, Any] = {
+            "result": Sessions().run(
+                filter, team, session_type=self.request.GET.get("session"), offset=self.request.GET.get("offset")
             )
-
-        if not date_filter.get("timestamp__lte"):
-            date_filter["timestamp__lte"] = now()
-
-        events = self.get_queryset()
```
Comment (EDsCODE) [inline posthog/api/event.py:302]: property filtering seems to have no effect now. originally it was wrapped into the processing for `self.get_queryset()` and now i don't believe it's handled anywhere in sessions

File: posthog/queries/sessions.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        time_series_avg = cursor.fetchall()
+
+        date_range = pd.date_range(filter.date_from, filter.date_to, freq=interval_freq,)
+        df = pd.DataFrame([{"date": a[0], "count": a[1], "breakdown": "Total"} for a in time_series_avg])
+        if interval == "week":
+            df["date"] = df["date"].apply(lambda x: x - pd.offsets.Week(weekday=6))
+        elif interval == "month":
+            df["date"] = df["date"].apply(lambda x: x - pd.offsets.MonthEnd(n=0))
+
+        df_dates = pd.DataFrame(df.groupby("date").mean(), index=date_range)
```
Comment (EDsCODE) [inline posthog/queries/sessions.py:200]: compare isn't working when it reaches this line. This also appears to have been broken before prior to this PR

**Review by EDsCODE — CHANGES_REQUESTED** [review]
Few issues

File: posthog/api/event.py (hunk 1 above)
Comment (timgl) [reply posthog/api/event.py:302]: Fixed

File: posthog/queries/sessions.py (hunk 2 above)
Comment (timgl) [reply posthog/queries/sessions.py:207]: Fixed, was actually an issue with empty Event querysets in general

**Comment by timgl** [comment]
Ready for another look @EDsCODE !

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**